packages/react-router-dom/rollup.config.js

```diff
@@ -0,0 +1,296 @@
+        sourcemap: !PRETTY,
+        banner: createBanner("React Router DOM", version),
+        globals: {
+          history: "HistoryLibrary",
+          "@remix-run/router": "Router",
+          react: "React",
+          "react-router": "ReactRouter",
+        },
+        name: "ReactRouterDOM",
+      },
+      external: ["history", "react", "react-router", "@remix-run/router"],
+      plugins: [
+        extensions({ extensions: [".ts", ".tsx"] }),
+        babel({
+          babelHelpers: "bundled",
+          exclude: /node_modules/,
+          presets: [
+            ["@babel/preset-env", { loose: true }],
+            "@babel/preset-react",
+            "@babel/preset-typescript",
+          ],
+          plugins: ["babel-plugin-dev-expression"],
+          extensions: [".ts", ".tsx"],
+        }),
+        replace({
+          preventAssignment: true,
+          values: { "process.env.NODE_ENV": JSON.stringify("development") },
+        }),
+      ].concat(PRETTY ? prettier({ parser: "babel" }) : []),
+    },
+    {
+      input: `${SOURCE_DIR}/index.tsx`,
+      output: {
+        file: `${OUTPUT_DIR}/umd/react-router-dom.production.min.js`,
+        format: "umd",
+        sourcemap: !PRETTY,
+        banner: createBanner("React Router DOM", version),
+        globals: {
+          history: "HistoryLibrary",
+          "@remix-run/router": "Router",
+          react: "React",
+          "react-router": "ReactRouter",
+        },
+        name: "ReactRouterDOM",
+      },
+      external: ["history", "react", "react-router", "@remix-run/router"],
+      plugins: [
+        extensions({ extensions: [".ts", ".tsx"] }),
+        babel({
+          babelHelpers: "bundled",
+          exclude: /node_modules/,
+          presets: [
+            ["@babel/preset-env", { loose: true }],
+            "@babel/preset-react",
+            "@babel/preset-typescript",
+          ],
+          plugins: ["babel-plugin-dev-expression"],
+          extensions: [".ts", ".tsx"],
+        }),
+        replace({
+          preventAssignment: true,
+          values: { "process.env.NODE_ENV": JSON.stringify("production") },
+        }),
+        // compiler(),
+        terser(),
+      ].concat(PRETTY ? prettier({ parser: "babel" }) : []),
+    },
+  ];
+
+  // Node entry points
+  const node = [
+    {
+      input: `${SOURCE_DIR}/node-main.js`,
+      output: {
+        file: `${OUTPUT_DIR}/main.js`,
+        format: "cjs",
+        banner: createBanner("React Router DOM", version),
+      },
+      plugins: [].concat(PRETTY ? prettier({ parser: "babel" }) : []),
+    },
+    {
+      input: `${SOURCE_DIR}/server.tsx`,
+      output: {
+        file: `${OUTPUT_DIR}/server.js`,
+        format: "cjs",
+      },
+      external: [
+        "url",
+        "history",
+        "react",
+        "react-dom/server",
+        "react-router-dom",
+        "@remix-run/router",
+      ],
+      plugins: [
+        extensions({ extensions: [".ts", ".tsx"] }),
```

Comment:
I don't see the TS plugin for `server`, does your config output type defs to the package root alongside the module? I *think* you'll need that for TS to find the module.

Reply:
yeah this first one seems to pick them all up:

<img width="257" alt="Screen Shot 2022-06-09 at 1 35 29 PM" src="https://user-images.githubusercontent.com/1609022/172909438-dc83bf93-18c6-4249-9893-ff6415cb100d.png">

Reply:
good catch - fixed in [69d25a1](https://github.com/remix-run/react-router/pull/8953/commits/69d25a148ad9b127ec8dedfc6f30a9bc84e39cd9)